controllers/argocd/rolebinding.go

```diff
@@ -312,13 +312,61 @@ func (r *ReconcileArgoCD) reconcileRoleBinding(name string, rules []v1.PolicyRul
 	return nil
 }
 
-func getCustomRoleName(name string) string {
-	if name == common.ArgoCDApplicationControllerComponent {
+// getCustomRoleName returns value of env variables defined for custom cluster role for namespace scoped Argo CD
+func getCustomRoleName(componentName string, cr *argoproj.ArgoCD) string {
+	// return env variable value for application-controller
+	if componentName == common.ArgoCDApplicationControllerComponent {
+		// Check if env variable is set in Argo CD resource
+		if cr.Spec.Controller.Env != nil {
+			for _, env := range cr.Spec.Controller.Env {
+				if env.Name == common.ArgoCDControllerClusterRoleEnvName {
+					return env.Value
+				}
+			}
+		}
+
+		// DEPRECATED: Check if env variable is set in Subscription resource
 		return os.Getenv(common.ArgoCDControllerClusterRoleEnvName)
```

Comment:
Do we want to deprecate subscription way of setting roles? I see value in having both via CR and subscription.

Reply:
Yes, it is changed now, there is also change in implementation, PTAL.